sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/optimizer/PropagateEmptyRelationSuite.scala

```diff
@@ -67,6 +68,28 @@ class PropagateEmptyRelationSuite extends PlanTest {
     comparePlans(optimized, correctAnswer)
   }
 
+  test("SPARK-32241: remove empty relation children from Union") {
```

Comment:
can we add a test to check the nullability update?

Reply:
A simple test case like here: https://github.com/apache/spark/pull/29053/commits/b454cd71fbd69bb0edf5113dbc0e72d70db5a6fd is ok? Or you want me to test that `UpdateAttributeNullability` kicks in and fixes some nullability after `PropagateEmptyRelation`?

Reply:
This test case does not go to the UpdateAttributeNullability rule in Optimizer, do you need to add a test case?